matchbox_signaling/src/signaling_server/mod.rs

```diff
@@ -0,0 +1,40 @@
+pub(crate) mod auth;
+pub(crate) mod builder;
+pub(crate) mod callbacks;
+pub(crate) mod error;
+pub(crate) mod handlers;
+pub(crate) mod server;
+
+use self::{auth::AuthKey, handlers::WsUpgradeMeta};
+use axum::response::Response;
+
+pub use server::SignalingServer;
+
+/// State managed by the signaling server
+pub trait SignalingState: Clone + Send + Sync + 'static {}
+
+/// Callbacks used by the signaling server
+pub trait SignalingCallbacks: Default + Clone + Send + Sync + 'static {}
+
+/// Supported Authentication types used by the signaling server
+pub trait Authentication: 'static {
+    fn verify(meta: WsUpgradeMeta, auth_key: AuthKey) -> Result<bool, Response>;
+}
+
+/// No-Op signaling callbacks
+#[derive(Default, Debug, Clone)]
+pub struct NoOpCallouts {}
+impl SignalingCallbacks for NoOpCallouts {}
+#[allow(unsafe_code)]
+unsafe impl Send for NoOpCallouts {}
+#[allow(unsafe_code)]
+unsafe impl Sync for NoOpCallouts {}
```

Comment:
These `#[allow(unsafe_code)]`s still need some explanation.

Reply:
As far as I can tell, it all derives from the use of `Rc<dyn Fn(In) -> Out>` in `Callback`. @simbleau Is there a reason you've chosen to go this way instead of an `Arc<Mutex<dyn Fn(In) -> Out>>` or likewise?